app/facet_selector/facet_selector.controller.js

```diff
@@ -35,11 +37,20 @@
       }
 
       $scope.set_date_filter = function(start_date, end_date) {
+        if (!start_date && !end_date) {
+          return;
+        }
+
         var default_start_date = -62167219200000; // BC 1
         var default_end_date = 3093496473600000; // AD 99999
         start_date = date_is_valid(start_date) ? Date.parse(start_date) : default_start_date;
         end_date = date_is_valid(end_date) ? Date.parse(end_date) : default_end_date;
 
+        // Can occur if either date is invalid, for example 2015/99/99
+        if (isNaN(start_date) || isNaN(end_date)) {
+          return;
```

Comment:
This seems to give some weird behavior for the filter updating/not when dates are partly valid/invalid. eg Type in start date 2000(creates `2000-`)-1-1, then type in end date 2010(creates `-2010`), which mysteriously drops the first date from the filter (because it's invalid).